src/ripple/consensus/Validations.h

```diff
@@ -719,21 +727,35 @@ class Validations
         std::lock_guard lock{mutex_};
         if (toKeep_)
         {
-            for (auto i = byLedger_.begin(); i != byLedger_.end(); ++i)
+            // We only need to refresh the keep range when it's just about to
+            // expire. Track the next time we need to refresh.
+            static std::chrono::steady_clock::time_point refreshTime;
+            if (auto const now = byLedger_.clock().now(); refreshTime <= now)
             {
-                auto const& validationMap = i->second;
-                if (!validationMap.empty() &&
-                    validationMap.begin()->second.seq() >= toKeep_)
+                // The next refresh time is shortly before the expiration
+                // time from now.
+                refreshTime = now + parms_.validationSET_EXPIRES -
+                    parms_.validationFRESHNESS;
+
+                for (auto i = byLedger_.begin(); i != byLedger_.end(); ++i)
                 {
-                    byLedger_.touch(i);
+                    auto const& validationMap = i->second;
+                    if (!validationMap.empty())
+                    {
+                        auto const seq = validationMap.begin()->second.seq();
+                        if (seq >= toKeep_->low_ && seq < toKeep_->high_)
+                        {
+                            byLedger_.touch(i);
+                        }
+                    }
                 }
-            }
 
-            for (auto i = bySequence_.begin(); i != bySequence_.end(); ++i)
-            {
-                if (i->first >= toKeep_)
+                for (auto i = bySequence_.begin(); i != bySequence_.end(); ++i)
                 {
-                    bySequence_.touch(i);
+                    if (i->first >= toKeep_->low_ && i->first < toKeep_->high_)
```

Comment:
Same style suggestion here re: the symmetry of the operators.

Reply:
Thanks, changed.